src/glean/private/event.h

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef EVENT_METRIC_H
+#define EVENT_METRIC_H
+
+#include "vpnglean.h"
+#if defined(UNIT_TEST)
+#  include "glean/glean.h"
+#  include <QJsonArray>
+#  include <QJsonDocument>
+#endif
+
+template <class T>
+class EventMetric final {
+ public:
+  constexpr explicit EventMetric(int aId) : m_id(aId) {}
+
+  void record() const { glean_event_record_no_extra(m_id); }
+
+  void record(T extras) const {
+    auto ffiExtras = extras.ToFfiExtra();
+    glean_event_record(m_id, ffiExtras.keys, ffiExtras.values, ffiExtras.count);
+  }
+
+#if defined(UNIT_TEST)
+  int32_t testGetNumRecordedErrors(Glean::ErrorType errorType) const {
+    return glean_event_test_get_num_recorded_errors(
+        m_id, static_cast<int32_t>(errorType));
+  }
+
+  QJsonArray testGetValue() const { return testGetValueInternal(QString()); }
+
+  QJsonArray testGetValue(const QString& pingName) const {
+    return testGetValueInternal(pingName);
+  }
```

Comment:
I think you can consolidate this to 
```suggestion
  QJsonArray testGetValue(const QString& pingName = QString()) const {
    return testGetValueInternal(pingName);
  }
```

Reply:
(and then you can fold `testGetValueInternal`'s body in too, if you wanna

Reply:
AAA, of course. I forget this is allowed in C++!